flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/controller/FlinkDeploymentController.java

```diff
@@ -169,4 +181,14 @@ private Optional<String> validateDeployment(FlinkDeployment deployment) {
         }
         return validationError;
     }
+
+    private void addMetricsIfAbsent(FlinkDeployment flinkApp) {
```

Comment:
We could call this methid `getMetrics` and simply use it everywhere instead of manuall getting from the map. That would potentially avoid some freak race conditions and null pointers between adding and getting.

Reply:
fixed